app/util/route.js

```diff
@@ -0,0 +1,89 @@
+import {equals} from 'ramda'
+import PreactRouter from 'preact-router'
+import {map, reduce} from 'wasmuth'
+import compose from '/util/compose'
+import {set, dispatch, getState} from '/store'
+import {routes} from '/routes'
+
+const {cloneElement} = Preact
+
+/**
+ * Use instaed of preact-router Router component.
+ * All children should be Route components (defined below).
+ * - Add key prop to Route children so that when route changes,
+ *   the current route component will unmount and the new one will mount.
+ * - Add path prop to Route children based on their name props.
+ */
+export const Router = ({children, routes, ...props}) =>
+  <PreactRouter {...props}>
+    {map((route, i) => cloneElement(route, {
+      key: i,
+      path: routes[route.attributes.name].path
+    }), children)}
+  </PreactRouter>
+
+/**
+ * Add preact-router props into the atom state
+ */
+export const Route = compose({
+  componentWillMount () {
+    const currentValues = getState().route
+    const newValues = {
+      args: this.props.matches,
+      url: window.location.pathname,
+      name: this.props.name
+    }
+    if (!equals(currentValues, newValues)) {
```

Comment:
Should we remove the duplicated functions in wasmuth and just use Ramda for functional utils? Should we even have wasmuth for common functions we use in each project? 

Maybe request.js should be it's own lib. Curious what you think about this.

Reply:
Hmm, wasmuth has been perfect for most utils I need, it's just on the rare occasion that I need something from ramda. If ramda isn't too big I'd probably prefer than + a requests lib.

Reply:
Ok cool, just use both libs for now.